xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -15148,14 +15153,17 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
         ai_list_tmp2 = pre_list_tmp1;
     }
     free(pre_list);
-    cds_list_for_each_entry(ai_list_tmp1, &ai_list->list, list)
-    {
-        if (ai_list_tmp1->info)
-            freeaddrinfo(ai_list_tmp1->info);
-        free(ai_list_tmp2);
-        ai_list_tmp2 = ai_list_tmp1;
+
+    if (ai_list != NULL) {
+        cds_list_for_each_entry(ai_list_tmp1, &ai_list->list, list)
+        {
+            if (ai_list_tmp1->info)
+                freeaddrinfo(ai_list_tmp1->info);
+            free(ai_list_tmp2);
+            ai_list_tmp2 = ai_list_tmp1;
+        }
+        free(ai_list);
     }
-    free(ai_list);
     free(ai_list_tmp2);
```

Comment:
you could have moved this one also inside "if"

Reply:
@aspandey Maybe I'm mistaken, but what if ai_list_tmp2 is being assigned a value at line 15153 but the condition in line 15157 will be false, so if the free will be moved inside the loop, it won't be freed. Can you double check please?

Reply:
Your point is valid. However, I would like to free the ai_list_tmp2 assigned on 15153 before this "if" only. Actually, we should also do this "if" for 15148 although it i might not have come in coverity scan.
It will keep the code more organized and less error prone, nothing more than that.

Reply:
I understood your intention. Some confusion was caused due to this being split into 2 patches, so combined them to 1.
Could you please give another look?